cpp/src/arrow/compute/exec/asof_join_node.cc

```diff
@@ -803,12 +805,25 @@ class CompositeReferenceTable {
   }
 
   template <class Type, class Builder = typename TypeTraits<Type>::BuilderType>
-  enable_if_fixed_width_type<Type, Status> static BuilderAppend(
+  enable_if_boolean<Type, Status> static BuilderAppend(
       Builder& builder, const std::shared_ptr<ArrayData>& source, row_index_t row) {
     if (source->IsNull(row)) {
       builder.UnsafeAppendNull();
       return Status::OK();
     }
+    builder.UnsafeAppend(bit_util::GetBit(source->template GetValues<uint8_t>(1), row));
```

Comment:
Not sure I understand here. Can you explain how this works?

Reply:
Can you remind me what is the constant "1" here again?

Reply:
The constant 1 selects the data buffer of the Arrow array. See [this doc](https://arrow.apache.org/docs/cpp/arrays.html).